### PR TITLE
School Library Milestone 6 : Preserve data in JSON files

### PR DESCRIPTION
**Pull request by Nemwel-Boniface**
## Hello reviewer(s) :wave: 
This is @selma-belhadj  @sangwa7 and @Nemwel-Boniface  team 
We are a group of three, but for some reason, two of us were able to work on the project. Our other team member had some other engagements and was not able to work with us on this milestone.

In this milestone we were able to:

- [x] Successfully save and retrieve data from JSON files
- [x] Check the documentation for Ruby's JSON library.
- [x] On exit from the app saves all the data (books, people, and rentals) into JSON files (`books.json`, `persons.json`, `rentals.json`) respectively.
- [x] On startup of the app load all data from files.
- [x] handle the case when one or more of the files are missing
- [x] Check and fix all found offenses

Our project has a bug we have not been able to work out. When you add a rental and exit the program and rejoin again, and try to list the books by a particular person, it generates a new id for the person and does not list the books rented by a specific person correctly. If you may be able to find the bug or know how to resolve it we will appreciate the input.

We believe the bug most probably comes from generating `id` as random from person.rb file `@id = Random.rand(1..99_999)` each time we call the initialize method of `Person` class the id is generated again.

The program will however work as required when you take the id of the person from the `rentals.json` file and use it as input for the "`Select a person from the list by its index: `" to list the rentals.

Thank you for taking the time to review our project :pray:

### CODE REVIEW

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Nemwel-Boniface and @selma-belhadj,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### This is a group of 3 but I cannot find commits from the 3rd partner. Kindly ensure there are contributions from all partners in the group

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: apps/rentals_app.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @persons = persons
+    @rentals = JSON.parse(File.read('./data/rentals.json'))
+  end
+
+  def add_rental
+    puts '********** Create a new rental **********'
+    puts 'Select which book you want to rent by entering its index: '
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the list by its index: '
+    person_index = gets.chomp.to_i
```
Comment (brytebee) [inline apps/rentals_app.rb]: - [x] **Optional:** While a user attempts to create a rental, consider displaying the available options to help them.

File: app.rb [hunk 2]
```diff
@@ -1,128 +1,67 @@
...
-      student = Student.new(age, classroom, name, parent_permission: false)
+      @persons.list_persons
+    when '3'
+      @persons.add_person
+    when '4'
+      @books.add_book
+    when '5'
+      @rentals.add_rental
+    when '6'
+      list_filtred_rentals
```
Comment (brytebee) [inline app.rb]: - [x] **Required:** When I tried to view the rentals before exiting the app, it crashes. Kindly look into this behaviour.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi Team 👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- On exit from the app the data is being saved.✔️
- On startup of the app load all data from files.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (selma-belhadj) [reply app.rb]: Thank you @brytebee for your review, I have made the required changes.

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @Nemwel-Boniface @selma-belhadj     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**